railslink/railslink
# add geo location information to membership submissions

This patch adds an `ip_address` field to the SlackMembershipSubmission
model which gets set to the clients IP address upon form submission. It
is then displayed in the admin screens when reviewing submissions
showing the city, region, and country.

It utilizes the free service from http://ip-api.com

test plan:
  - % bundle install
  - % rails s

  - delete any pending membership submissions you have locally

      % rails runner SlackMembershipSubmission.delete_all

  - visit home page
  - click 'join now' button
  - fill out form and submit
  - go to /admin
  - go to pending submissions
  - note that under the 'Location' field there isn't any additional
    text.
  - open rails console and tweak your submission to give it a public ip
    address:

      % rails c
      > sms = SlackMembershipSubmission.first
      > sms.ip_address = "73.11.237.17"
      > sms.save!

  - reload pending submissions screen
  - note that under 'Location' field it now says 'Olympia, WA, United
    States'

<img width="471" alt="Screen Shot 2019-03-23 at 3 26 50 PM" src="https://user-images.githubusercontent.com/16705/54872363-1d0c2a80-4d80-11e9-96b3-2267c1405634.png">
